services/streammanager/kafka/kafkamanager.go

```diff
@@ -92,7 +92,9 @@ func NewProducer(destinationConfig interface{}) (sarama.SyncProducer, error) {
 	var destConfig = Config{}
 	jsonConfig, err := json.Marshal(destinationConfig)
 	err = json.Unmarshal(jsonConfig, &destConfig)
-
+	if err != nil {
+		pkgLogger.Errorf("Error while unmarshalling dest config %v", err)
```

Comment:
Lets wrap this error and return it here.

Reply:
done